simulator/core/perf_sim.cpp

```diff
@@ -145,63 +145,96 @@ void PerfMIPS::clock_decode( int cycle)
 
     /* receive flush signal */
     const bool is_flush = rp_decode_flush->is_ready( cycle) && rp_decode_flush->read( cycle);
-
+    IfIdData _data;
     /* branch misprediction */
     if ( is_flush)
     {
         /* ignoring the upcoming instruction as it is invalid */
-        decode_data = rp_fetch_2_decode->read( cycle);
+        _data = rp_fetch_2_decode->read( cycle);
+	    if(rp_decode_2_decode->is_ready(cycle))
```

Comment:
We do not use tabs for indentation. Please change settings of your editor (we have a cheatsheet for Vim and Notepad++ on Wiki)